framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -132,6 +132,9 @@
     "assumeAssertionsAreEnabled",
     "assumeAssertionsAreDisabled",
 
+    // Whether to assume receiver and params of the method to be non-null after the method
+    // invocation.
+    "receiverAndParamsNullableAfterMethodCall",
```

Comment:
We should think of a name that will allow us to, in the future, change the default behavior to be conservative and have the option be the more optimistic behavior.
How about `conservativeArgumentNullnessAfterMethodCall`?
- `conservative` will allow us to change it to `optimistic` in the future.
- `Argument` the receiver is just another argument (and arguments, not parameters).
- `Nullness`, not `Nullable`, as some things will stay non-null.
- `AfterMethodCall` is a bit long... Why is this only relevant for method calls, not also constructor invocations? For symmetry, shouldn't the same apply there? Can you write a test? If so, maybe `AfterInvocation`?

Reply:
I can't think of a way to write such tests. When invoking the constructor, we don't need a receiver, and I can not assign null to the receiver inside the constructor. For method parameters of the constructor invocation, there's no way to make them become null afterwards.
For example
```java
import org.checkerframework.checker.nullness.qual.Nullable;
class Issue300D {
    @Nullable Object f = "";
    Issue300D() {}
    Issue300D(Object o) {
        f = null; // this.f assigned to null, nothing to do with r.f
        o = null // reference o assigned to null, nothing to do with r.f
    }
    public static void main(String[] args) {
        Issue300D r = new Issue300D();
        if (r.f == null) {return;}

        new Issue300D(r.f);
        r.f.toString();
    }
}
```